src/content/plugins.md

```diff
@@ -0,0 +1,325 @@
+when the `needsCache` option is true, and the directory will only be deleted when the Broccoli 
+process exits.
+
+All paths stay the same between rebuilds.
+
+# Example plugin
+
+Let's build a sample concatenation plugin. It's going to concatenate a set of files matching a glob expression.
+
+```js
+import Plugin from 'broccoli-plugin';
+import walkSync from 'walk-sync';
+import fs from 'fs';
+
+export class ConcatPlugin extends Plugin {
+    constructor(inputNodes, options) {
+        super(inputNodes, options);
+        
+        this.fileMatchers = options.globs || ['**/*'];
+        this.joinSeparator = options.joinSeparator || "\n";
+        this.outputFile = options.outputFile || 'concat';
+    }
+    
+    build() {
+        const walkOptions = {
+            includeBasePath: true,
+            directories: false,
+            globs: this.fileMatchers,
+        };
+
+        const content = this.inputPaths
+            .reduce((output, inputPath) => output + this.joinSeparator +
+                walkSync(inputPath, walkOptions)
+                    .map(file => fs.readFileSync(file, { encoding: 'UTF-8' }))
+                    .join(this.joinSeparator),
+            '');
+
+        fs.writeFileSync(`${this.outputPath}/${this.outputFile}`, content);
+    }
+}
+
+export default function concatPlugin(...params) {
+    return new ConcatPlugin(...params);
+}
+```
+
+Let's take a look and see what's happening here. 
+
+First, we are importing our dependencies. We are using 3 packages, the
+[broccoli-plugin](https://github.com/broccolijs/broccoli-plugin) base class that we are extending, a package 
+called [walk-sync](https://www.npmjs.com/package/walk-sync) which synchronously walks a directory recursively, 
+and the Node [fs](https://nodejs.org/api/fs.html) package.
+
+After this we define the class and the constructor:
+```js
+export class ConcatPlugin extends Plugin {
+    constructor(inputNodes, options) {
+        super(inputNodes, options);
+        
+        this.fileMatchers = options.globs || ['**/*'];
+        this.joinSeparator = options.joinSeparator || "\n";
+        this.outputFile = options.outputFile || 'concat';
+    }
+```
+
+Here we are defining our class and exporting it so that other plugins can extend it by doing:
+```js
+import { ConcatPlugin } from 'concat-plugin';
+
+class MyPlugin extends ConcatPlugin {
+```
+
+The constructor accepts multiple inputNodes and an options hash. As you can see we are defining 3 options, a 
+default glob expression, the join character and the output file name. These will be used to inform how our build
+method should work. 
+
+Next up we define our `build()` method. First we setup some options for the walk-sync package then we iterate 
+`this.inputPaths` using a reduce function. If you've not used reduce before, it provides a simple functional
+programming way of iterating an input and combining the output into an accumulator. In our case we are merely 
```

Comment:
I greatly appreciate the thoughtfulness of this aside on reduce. What would you say is the target audience for Broccoli? It seems likely to me that we could assume knowledge of reduce, in which case this might be speaking down to folks a tad?

Reply:
TBH the target is any ember developer. I'm not sure if `reduce` is widely known. I feel like `if you've not used reduce before` caters for those who don't know and a skip-over for those who do.